gwsurrogate/surrogate.py

```diff
@@ -2282,6 +2282,76 @@ def _get_intrinsic_parameters(self, q, chiA0, chiB0, precessing_opts,
+    """
+A class for the NRHybSur3dq8_CCE surrogate model presented in FIXME,
+arxiv:FIXME.
+
+Evaluates gravitational waveforms generated by aligned-spin binary black hole
+systems. This model was built using CCE numerical relativity (NR) waveforms that
+have been hybridized using post-Newtonian (PN) and effective one body (EOB)
+waveforms.
+
+This model includes the following spin-weighted spherical harmonic modes:
+(2,2), (2,1), (2,0), (3,3), (3,2), (3,0), (4,4) (4,3), (4,0), and (5,5).
+The m<0 modes are deduced from the m>0 modes.
+
+The parameter space of validity is:
+q \in [1, 10] and chi1z/chi2z \in [-1, 1],
+where q is the mass ratio and chi1z/chi2z are the spins of the heavier/lighter
+BH, respectively, in the direction of orbital angular momentum.
+
+The surrogate has been trained in the range
+q \in [1, 8] and chi1z/chi2z \in [-0.8, 0.8], but produces reasonable waveforms
+in the above range and has been tested against existing NR waveforms in that
+range.
+
+See the __call__ method on how to evaluate waveforms.
+   """
+
+    def __init__(self, h5filename):
+        self.h5filename = h5filename
+        domain_type = 'Time'
+        keywords = {
+            'Precessing': False,
+            'Hybridized': True,
+            }
+        # soft_lims -> raise warning when outside lims
+        # hard_lim -> raise error when outside lims
+        # Format is [qMax, chiMax].
+        soft_param_lims = [8.01, 0.801]
+        hard_param_lims = [10.01, 1]
+        super(NRHybSur3dq8_CCE, self).__init__(self.__class__.__name__, \
+            domain_type, keywords, soft_param_lims, hard_param_lims)
+
+    def _load_dimless_surrogate(self):
+        """
+        This function, which must be overriden for each derived class of
+        SurrogateEvaluator, handles the loading of the dimensionless surrogate.
+        This should return the loaded surrogate.
+        The loaded surrogate should have a __call__ function that returns the
+        dimensionless time/frequency array and dimensionless waveform modes.
+        The return value of this functions will be stored as
+        self._sur_dimless()
+        The __call__ function of self._sur_dimless() should take all inputs
+        passed to self._sur_dimless() in the __call__ function of this class.
+        """
+        sur = new_surrogate.AlignedSpinCoOrbitalFrameSurrogate()
+        sur.load(self.h5filename)
+        return sur
+
+    def _get_intrinsic_parameters(self, q, chiA0, chiB0, precessing_opts,
+            tidal_opts, par_dict):
+        """
+        This function, which must be overriden for each derived class of
+        SurrogateEvaluator, puts all intrinsic parameters of the surrogate
+        into a single array.
+        For example, for NRHybSur3dq8: x = [q, chiAz, chiBz].
+        """
+        if par_dict is not None:
+            raise ValueError('Expected par_dict to be None.')
+        x = [q, chiA0[2], chiB0[2]]
+        return x
```

Comment:
As far as I can tell, this is identical to `NRHybSur3dq8` except for (i) the docstring, and (ii) the call to `super()` on line 2324. To avoid repeating code, can't we just inherit from NRHybSur3dq8, only setting a new docstring?

Reply:
I guess I can change to 
`class NRHybSur3dq8_CCE(NRHybSur3dq8):`
`      (docstring)`
`      pass`
But, should I keep the `__init__`so that we can modify `soft_param_lims` and `hard_param_lims` 
for this particular model later, if necessary?

Reply:
Yes, I didn't know that the limits might change

Reply:
I think I can always add the bit later if it needs to be changed; so I just removed it.